### ISSUE TITLE
Config file bug. Single item list interpreted as the item itself.

### ISSUE THREAD

**Issue by mattangus** [guildai/guildai#271]
Here is a self contained example to reproduce:

guild.yml
```
test:
  main: test
  flags-dest: config:test.yaml
  flags-import: all
```

test.py
```
import yaml

with open("test.yaml") as f:
    params = yaml.full_load(f)

print(params["a_list"])
```

test.yaml
```
a_list:
- single value
```
or
```
a_list: [single value]
```

### Guild output

```
$ guild run test -y
Resolving config:test.yaml dependency
'single value'
```

### Python output

```
$ python3 test.py 
['single value']
```

The bug here is that these two disagree. When I'm coding I test my scripts withoout guild first. For example I might not know all the parameters to an experiment or what source code is invlolve (all things required to set up a proper `guild.yml`). But having different behaviour between running it directly and through guild is a code breaking issue.

The only work around I can think of is to add it to a list if the type of the loaded value is not a list. Which essentially ammounts to maintaining two versions of the code. One for guild, one for regular python execution. In my case this should be fine but could be prohibatively expensive to convert code if there are many parameters like this.

Currently on guild version `0.7.2`

## Update
After investigating other workarounds. Lists all together seem to be broken.

Changing `test.yaml` to:
```
a_list: 
  - single value
  - second value
```

And the output is:
```
$ guild run test -y
Resolving config:test.yaml dependency
'single value' 'second value'
$ python3 test.py
['single value', 'second value']
```

Guild seems to convert the list to a string space separated with single quotes around each element.

**Comment by gar1t** [comment]
Great report - thanks for the detail! I created a resolution doc here:

https://github.com/guildai/issue-resolution/tree/master/271-config-file-bug-single-item-list-interpreted-as-the-item-itself-

There's unfortunately no good workaround at this time. The above resolution doc proposes using a string to encode a list but that's disruptive to your code, which we want to avoid.

I'll update here with progress.

**Comment by gar1t** [comment]
This is resolved in master and will be available in the next pre-release (0.7.3.rc2). We can keep this issue open until you have a chance to verify. Thanks again!

**Comment by mattangus** [comment]
Great! any idea when rc2 will hit pypi? I can see rc1 is already on there. I'll install from source for now!

**Comment by mattangus** [comment]
Just installed from source (master branch) and I can confirm this is fixed.

**Comment by gar1t** [comment]
Great! This will be available in 0.7.3, which is due out today or tomorrow. Closing but if there's a regression we can reopen. Thanks again!
